--- COLLADABaseUtils/include/COLLADABUPlatform.h
+++ COLLADABaseUtils/include/COLLADABUPlatform.h
@@ -12,7 +12,7 @@
 #define __COLLADABU_PLATTFORM_H__
 
 
-#if (defined(__APPLE__) || defined(OSMac_)) && (defined(__GNUC__) || defined(__xlC__) || defined(__xlc__)) || defined(__APPLE_CC__) 
+#if (defined(__APPLE__) || defined(OSMac_)) && (defined(__GNUC__) || defined(__xlC__) || defined(__xlc__)) || defined(__APPLE_CC__) || defined(__FreeBSD__)
 #  define COLLADABU_OS_MAC
 #  ifdef __LP64__
 #    define COLLADABU_OS_MAC64
