--- a/libltdl/config/ltmain.sh	2013-07-04 17:33:18.246125827 +0200
+++ b/libltdl/config/ltmain.sh	2013-07-04 17:34:17.981126321 +0200
@@ -7896,13 +7896,15 @@ func_mode_link ()
 	  ;;
 
 	freebsd-aout)
-	  major=".$current"
-	  versuffix=".$current.$revision";
+	  func_arith $current - $age
+	  major=.$func_arith_result
+	  versuffix="$major.$revision"
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current"
+	  func_arith $current - $age
+	  major=.$func_arith_result
+	  versuffix="$major"
 	  ;;
 
 	irix | nonstopux)
