--- identcpu.orig	2015-02-17 13:06:42.608560444 +0000
+++ identcpu.c	2015-02-17 13:10:06.416938088 +0000
@@ -84,7 +84,7 @@
 static void print_vmx_info(void);
 
 int	cpu_class;
-char machine[] = MACHINE;
+char machine[] = "x86_64";
 
 #ifdef __amd64__
 #ifdef SCTL_MASK32
@@ -95,7 +95,7 @@
 sysctl_hw_machine(SYSCTL_HANDLER_ARGS)
 {
 #ifdef SCTL_MASK32
-	static const char machine32[] = "i386";
+	static const char machine32[] = "i686";
 #endif
 	int error;
 
