Index: k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
===================================================================
diff -Naur k3b-2.0.2-orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-2.0.2-orig/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2014-09-15 14:35:16.345772461 -0400
+++ k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2014-09-15 14:36:38.645637478 -0400
@@ -165,7 +165,11 @@
     }
 
     if( d->formatContext ) {
-        ::av_close_input_file( d->formatContext );
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(53, 17, 0)
+      ::avformat_close_input(&d->formatContext);
+#else
+      ::av_close_input_file( d->formatContext );
+#endif
         d->formatContext = 0;
     }
 }
