diff -rup pyalpm-0.6.2/src/db.c new/src/db.c
--- src/db.c	2013-04-01 14:21:10.000000000 +0200
+++ src/db.c	2014-05-30 13:35:58.866687247 +0200
@@ -220,7 +220,7 @@ static struct PyMethodDef db_methods[] =
     "update a database from its url attribute\n"
     "args: force (update even if DB is up to date, boolean)\n"
     "returns: True if an update has been done" },
-  { NULL },
+  { NULL, NULL, 0, NULL }
 };
 
 struct PyGetSetDef db_getset[] = {
@@ -231,7 +231,7 @@ struct PyGetSetDef db_getset[] = {
     "a list of URLs (for sync DBs)", NULL } ,
   { "pkgcache", (getter)pyalpm_db_get_pkgcache, 0, "(read only) list of packages", NULL } ,
   { "grpcache", (getter)pyalpm_db_get_grpcache, 0, "(read only) list of package groups", NULL } ,
-  { NULL }
+  { NULL, NULL, 0, NULL, NULL }
 };
 
 static PyTypeObject AlpmDBType = {
diff -rup pyalpm-0.6.2/src/handle.c new/src/handle.c
--- src/handle.c	2013-04-01 14:21:10.000000000 +0200
+++ src/handle.c	2014-05-30 13:36:17.505685272 +0200
@@ -332,7 +332,7 @@ struct PyGetSetDef pyalpm_handle_getset[
     &cb_getsets[CB_PROGRESS] },
 
   /** terminator */
-  { NULL }
+  { NULL, NULL, 0, NULL, NULL }
 };
 
 static PyMethodDef pyalpm_handle_methods[] = {
diff -rup pyalpm-0.6.2/src/package.c new/src/package.c
--- src/package.c	2013-04-01 14:21:24.000000000 +0200
+++ src/package.c	2014-05-30 13:35:22.658689105 +0200
@@ -318,13 +318,13 @@ static struct PyGetSetDef AlpmPackageGet
   /* miscellaneous information */
   { "has_scriptlet", (getter)pyalpm_pkg_has_scriptlet, 0, "True if the package has an install script", NULL },
   { "download_size", (getter)pyalpm_pkg_download_size, 0, "predicted download size for this package", NULL },
-  { NULL }
+  { NULL, NULL, 0, NULL, NULL }
 };
 
 static struct PyMethodDef pyalpm_pkg_methods[] = {
   { "compute_requiredby", pyalpm_pkg_compute_requiredby, METH_NOARGS,
       "computes the list of packages requiring this package" },
-  { NULL }
+  { NULL, NULL, 0, NULL }
 };
 
 PyTypeObject AlpmPackageType = {
diff -rup pyalpm-0.6.2/src/transaction.c new/src/transaction.c
--- src/transaction.c	2013-04-01 14:21:10.000000000 +0200
+++ src/transaction.c	2014-05-30 13:36:36.881683875 +0200
@@ -418,7 +418,7 @@ static struct PyGetSetDef pyalpm_trans_g
   { "flags", (getter)pyalpm_trans_get_flags, NULL, "Transaction flags", NULL } ,
   { "to_add", (getter)pyalpm_trans_get_add, NULL, "Packages added by the transaction", NULL },
   { "to_remove", (getter)pyalpm_trans_get_remove, NULL, "Packages added by the transaction", NULL },
-  { NULL }
+  { NULL, NULL, 0, NULL, NULL }
 };
 
 static struct PyMethodDef pyalpm_trans_methods[] = {
@@ -437,7 +437,7 @@ static struct PyMethodDef pyalpm_trans_m
     "set the transaction to perform a system upgrade\n"
     "args:\n"
     "  transaction (boolean) : whether to enable downgrades\n" },
-  { NULL }
+  { NULL, NULL, 0, NULL }
 };
 
 /* The Transaction object have the same underlying C structure
