--- b/bleachbit/Common.py	        2015-06-20 00:03:56.000000000 -0400
+++ a/bleachbit/Common.py         2015-08-27 12:07:01.881265522 -0400
@@ -103,6 +103,8 @@
     system_cleaners_dir = os.path.join(bleachbit_exe_path, 'share\\cleaners\\')
 elif sys.platform[:6] == 'netbsd':
     system_cleaners_dir = '/usr/pkg/share/bleachbit/cleaners'
+elif sys.platform[:7] == 'freebsd':
+    system_cleaners_dir = '/usr/share/bleachbit/cleaners'
 else:
     system_cleaners_dir = None
     logger.warning('unknown system cleaners directory for platform %s ' %
@@ -135,6 +137,8 @@
         locale_dir = os.path.join(bleachbit_exe_path, 'share\\locale\\')
     elif sys.platform[:6] == 'netbsd':
         locale_dir = "/usr/pkg/share/locale/"
+    elif sys.platform.startswith('freebsd'):
+        local_dir = "/usr/share/locale/"


 # launcher
