diff -rupN cmus-v2.5.0/scripts/checks.sh src/cmus-v2.5.0/scripts/checks.sh
--- cmus-v2.5.0/scripts/checks.sh	2012-11-07 10:46:17.000000000 +0200
+++ src/cmus-v2.5.0/scripts/checks.sh	2013-09-11 14:17:17.000000000 +0300
@@ -125,8 +125,8 @@ cc_cxx_common()
 
 	case `uname -s` in
 	*BSD)
-		common_cf="$common_cf -I/usr/local/include"
-		common_lf="$common_lf -L/usr/local/lib"
+		common_cf="$common_cf -I/usr/include"
+		common_lf="$common_lf -L/usr/lib"
 		;;
 	Darwin)
 		# fink
