--- lib/verify.h.orig	2013-10-04 06:32:16.000000000 -0600
+++ lib/verify.h	2013-10-04 06:33:44.000000000 -0600
@@ -39,6 +39,7 @@
 # if (99 < __GNUC__) && defined __cplusplus
 #  define _GL_HAVE_STATIC_ASSERT 1
 # endif
+#undef _GL_HAVE__STATIC_ASSERT
 
 /* Each of these macros verifies that its argument R is nonzero.  To
    be portable, R should be an integer constant expression.  Unlike
@@ -194,6 +195,7 @@
    ordinary declaration, and cannot appear inside struct { ... }.  */
 
 # ifdef _GL_HAVE__STATIC_ASSERT
+#error
 #  define _GL_VERIFY _Static_assert
 # else
 #  define _GL_VERIFY(R, DIAGNOSTIC)				       \
