
 This seems to be lacking some symbols otherwise

--- a/GNUmakefile.in	2013-11-30 20:51:00.000000000 +0100
+++ a/GNUmakefile.in	2013-11-30 20:51:00.000000000 +0100
@@ -687,8 +687,9 @@
 Programs_WebKitPluginProcess_LDADD = \
 	-lpthread \
 	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 	WebCoreLayerGtk2.a \
 	libWebKit2Platform.la \
+	*.la \
 	$(CAIRO_LIBS) \
 	$(COVERAGE_LDFLAGS) \
 	$(FREETYPE_LIBS) \
