--- a/slimlock.cpp	2014-01-09 14:10:28.357059460 +0100
+++ b/slimlock.cpp	2014-01-09 14:12:40.233103051 +0100
@@ -12,7 +12,7 @@
 #include <algorithm>
 #include <sys/types.h>
 #include <sys/ioctl.h>
-#include <linux/vt.h>
+#include <sys/consio.h>
 #include <X11/keysym.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
@@ -191,7 +191,8 @@ int main(int argc, char **argv) {
 		if ((term = open("/dev/console", O_RDWR)) == -1)
 			perror("error opening console");
 
-		if ((ioctl(term, VT_LOCKSWITCH)) == -1)
+		int data = 1;
+		if ((ioctl(term, VT_LOCKSWITCH, &data)) == -1)
 			perror("error locking console");
 	}
 
@@ -250,7 +251,8 @@ int main(int argc, char **argv) {
 	close(lock_file);
 
 	if(cfg->getOption("tty_lock") == "1") {
-		if ((ioctl(term, VT_UNLOCKSWITCH)) == -1) {
+		int data = 2;
+		if ((ioctl(term, VT_LOCKSWITCH, &data)) == -1) {
 			perror("error unlocking console");
 		}
 	}
@@ -349,7 +351,8 @@ void HandleSignal(int sig)
 			DPMSDisable(dpy);
 	}
 
-	if ((ioctl(term, VT_UNLOCKSWITCH)) == -1) {
+	int data = 2;
+	if ((ioctl(term, VT_LOCKSWITCH, &data)) == -1) {
 		perror("error unlocking console");
 	}
 	close(term);
