--- libgksu/libgksu.c	2013-10-06 02:11:12.000000000 +0300
+++ ../b/libgksu/libgksu.c	2013-10-06 02:12:14.000000000 +0300
@@ -1981,7 +1981,7 @@ gksu_su_fuller (GksuContext *context,
       gchar **cmd = g_malloc (sizeof(gchar*)*7);
 
       setsid();   // make us session leader
-      cmd[i] = g_strdup ("/bin/su"); i++;
+      cmd[i] = g_strdup ("/usr/bin/su"); i++;
       if (context->login_shell)
 	{
 	  cmd[i] = g_strdup ("-"); i++;
@@ -2003,7 +2003,7 @@ gksu_su_fuller (GksuContext *context,
       if (execv (cmd[0], cmd) == -1)
 	{
 	  fprintf (stderr,
-		   "Unable to run /bin/su: %s",
+		   "Unable to run /usr/bin/su: %s",
 		   strerror(errno));
 	}
 
